### PR TITLE
feat(core): Add commands to workers to respond with current state

### PR DESCRIPTION
**Pull request by flipswitchingmonkey**
This PR adds new endpoints to the REST API: `/orchestration/worker/status` and `/orchestration/worker/id`

Currently these just trigger the return of status / ids from the workers via the redis back channel, this still needs to be handled and passed through to the frontend.

It also adds the eventbus to each worker, and triggers a reload of those eventbus instances when the configuration changes on the main instances.

### CODE REVIEW

File: packages/cli/src/controllers/orchestration.controller.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import config from '@/config';
+import { Authorized, Get, RestController } from '@/decorators';
+import { OrchestrationRequest } from '@/requests';
+import { Service } from 'typedi';
+import { OrchestrationService } from '../services/orchestration.service';
+
+@Authorized()
```
Comment (krynble) [inline packages/cli/src/controllers/orchestration.controller.ts:7]:
```suggestion
@Authorized(['global', 'owner'])
```

File: packages/cli/src/controllers/orchestration.controller.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+	@Get('/worker/status')
+	async getWorkersStatusAll() {
+		return this.orchestrationService.getWorkerStatus();
+	}
+
+	@Get('/worker/ids')
+	async getWorkerIdsAll() {
+		return this.orchestrationService.getWorkerIds();
+	}
```
[the remark this answers is not in the document]
Comment (krynble) [reply packages/cli/src/controllers/orchestration.controller.ts:29]: Also we should be adding integration tests to those endpoints

File: packages/cli/src/services/orchestration.service.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+		// TODO: these are all proof of concept implementations for the moment
+		// until worker communication is implemented
+		// #region proof of concept
+		await this.redisSubscriber.subscribeToEventLog();
+		await this.redisSubscriber.subscribeToWorkerResponseChannel();
+		await this.redisSubscriber.subscribeToCommandChannel();
+
+		this.redisSubscriber.addMessageHandler(
+			'OrchestrationMessageReceiver',
+			async (channel: string, messageString: string) => {
```
Comment (krynble) [inline packages/cli/src/services/orchestration.service.ts:54]: nit: break into smaller parts and add tests

File: packages/cli/src/controllers/orchestration.controller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/controllers/orchestration.controller.ts:29]: These endpoint currently do not return anything, they just trigger the messsage to the workers to respond on Redis with their status.
TODO: these responses need to be forwarded to and handled by the frontend

File: packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts [hunk 4]
```diff
@@ -70,7 +76,7 @@ export class MessageEventBus extends EventEmitter {
 	 *
 	 * Sets `isInitialized` to `true` once finished.
 	 */
-	async initialize() {
+	async initialize(options?: MessageEventBusInitializeOptions): Promise<void> {
```
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts:79]: We can, but I'd prefer to do that as a separate PR

File: packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts [hunk 5]
```diff
@@ -195,6 +213,12 @@ export class MessageEventBus extends EventEmitter {
 		return result;
 	}
 
+	async tellWorkersToUpdateDestinations() {
```
[the remark this answers is not in the document]
Comment (flipswitchingmonkey) [reply packages/cli/src/eventbus/MessageEventBus/MessageEventBus.ts:216]: renamed to broadcastRestartEventbusAfterDestinationUpdate

**Review by krynble — APPROVED** [review]
LGTM tested and working.

**Comment by janober** [comment]
Got released with [`n8n@1.7.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.7.1)

**Pull request merged**